UI/window-basic-main.cpp

```diff
@@ -410,6 +410,29 @@ void OBSBasic::UpdateVolumeControlsDecayRate()
 	}
 }
 
+void OBSBasic::UpdateVolumeControlsPeakMeterType()
+{
+	uint32_t peakMeterTypeIdx = config_get_uint(basicConfig, "Audio",
+			"PeakMeterType");
+
+	enum obs_peak_meter_type peakMeterType;
```

Comment:
Since `peakMeterTypeIdx` is a uint with only two possible values, and a default of `0`, would casting `peakMeterTypeIdx` to `obs_peak_meter_type` make more sense here?  I tried to find examples of either code style in the existing OBS codebase, but I've struggled to find applicable examples.

Reply:
I don't think it is proper code style to cast between integer and enum values. This code makes it very clear which integer values correspond with which enum value.

Also it is easy and clear how to handle integer values outside the enum range, without resorting to adding PEAK_METER_TYPE_END enum value and comparing against it. Which on modern compilers requires the PEAK_METER_TYPE_END to be used in switch statements on that enum.

Reply:
I couldn't speak to whether or not it is proper code style to cast between integer and enum, here or elsewhere.  The code style documents for this project don't mention anything specifically when casts are and aren't appropriate.  I know casts *are* used elsewhere in this project, and I know that enums are *sort of* ints.

Yes, I will agree that this makes it clear which peakMeterTypeIdx corresponds to which value for the obs_peak_meter_type enum.  It just seemed like a large chunk of code (used more than once) for what seems to be type juggling.

Reply:
I found an example in OBS code of type casting between int and enum:
https://github.com/obsproject/obs-studio/blob/189e535014287a7c4999e716c2edb5fc07244b19/libobs/obs.h#L980
https://github.com/obsproject/obs-studio/blob/61839d8c94466a3d762837f2e198d982ef5a1eb6/UI/adv-audio-control.cpp#L304
https://github.com/obsproject/obs-studio/blob/61839d8c94466a3d762837f2e198d982ef5a1eb6/UI/adv-audio-control.cpp#L98

OBSAdvAudioCtrl::monitoringTypeChanged seems like a good example.

Reply:
The reason we do it is because of Microsoft's compiler.  It's not particularly any fault of your own.  Microsoft's compiler warnings aren't the most useful.